android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTest.java

```diff
@@ -0,0 +1,1954 @@
+        List<EntitySet> entitySetList = createEntityList(2);
+        int[] itemsErrorIndexes = new int[] { 0, 1 };
+
+        DataStore<EntitySet> storeSync = DataStore.collection(EntitySet.COLLECTION, EntitySet.class, StoreType.SYNC, client);
+        clearBackend(storeSync);
+        client.getSyncManager().clear(EntitySet.COLLECTION);
+
+        DefaultKinveyClientListCallback saveListCallback = saveList(storeSync, entitySetList);
+        assertNotNull(saveListCallback.result);
+        assertTrue(checkIfItemsAtRightIndex(entitySetList, saveListCallback.result,
+                itemsErrorIndexes, EntitySet.DESCRIPTION_KEY, false, false));
+
+        DefaultKinveyPushCallback pushCallback = push(storeSync, LONG_TIMEOUT);
+        assertNotNull(pushCallback.error);
+        if (pushCallback.error instanceof KinveySaveBunchException) {
+            assertTrue(checkIfItemsAtRightIndex(entitySetList, ((KinveySaveBunchException) pushCallback.error).getEntities(),
+                    itemsErrorIndexes, EntitySet.DESCRIPTION_KEY, true, false));
+            assertTrue(checkErrorsIndexes(((KinveySaveBunchException) pushCallback.error).getErrors(), itemsErrorIndexes));
+        }
+
+        List<SyncItem> syncItems = pendingSyncEntities(EntitySet.COLLECTION);
+        assertNotNull(syncItems);
+        assertEquals(syncItems.size(), entitySetList.size());
+
+        DefaultKinveyReadCallback findCallback = find(storeSync, LONG_TIMEOUT);
+        assertNotNull(findCallback.result);
+        checkIfSameObjects(entitySetList, findCallback.result.getResult(), EntitySet.DESCRIPTION_KEY, false);
+    }
+
+    @Test
+    @Ignore("Currently fail because of KCS bug: https://kinvey.atlassian.net/browse/BACK-3959")
+    public void testPushItemsListReturnsErrorsForEachItemErrorSync() throws InterruptedException {
+        print("should return the failure reason in the result for each pushed item when they are different");
+        // create an array with 3 items without _id, two of which should be invalid for different reasons
+        // save using the array above
+        // push(), should return the items with respective POST and PUT operations and items which are set to fail should have their distinct errors
+        // pendingSyncEntities(), should return the failing items from step 1
+        // find using syncstore, should return the items from step 1 - the successfull one with lmt and ect
+        List<Person> personsList = createPushErrList();
+        int[] successItemsIdx = new int[] { 2 };
+
+        DataStore<Person> storeSync = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        clearBackend(storeSync);
+        client.getSyncManager().clear(Person.COLLECTION);
+
+        saveList(storeSync, personsList);
+
+        DefaultKinveyPushCallback pushCallback = push(storeSync, LONG_TIMEOUT);
+        assertNotNull(pushCallback.result);
+        assertEquals(personsList.size(), pushCallback.result.getSuccessCount());
```

Comment:
Yes, right. I fixed this. Found out that there also some bug in sdk for push items success counter.
It will fix this in my task of push logic updates for multi-insert.